tests/e2e/e2e_setup_test.go

```diff
@@ -326,3 +332,100 @@ func noRestart(config *docker.HostConfig) {
 		Name: "no",
 	}
 }
+
+func (s *IntegrationTestSuite) initUpgrade() {
+	for x := range s.chains {
+		c := s.chains[x]
+		s.T().Logf("submitting upgrade proposal for chain-id: %s", c.ChainMeta.Id)
+		cmdStr := fmt.Sprintf("docker exec %v bash -c 'chmod +x /scripts/osmosis_upgrade.sh && /scripts/osmosis_upgrade.sh \"%s\"'", s.valResources[c.ChainMeta.Id][0].Container.ID, c.ChainMeta.Id)
+		exec.Command("/bin/sh", "-c", cmdStr).Output()
+	}
+
+	for x := range s.chains {
+		c := s.chains[x]
+		s.T().Logf("waiting to reach upgrade height for chain-id: %s", c.ChainMeta.Id)
+		type status struct {
+			LatestHeight string `json:"latest_block_height"`
+		}
+
+		type syncInfo struct {
+			SyncInfo status `json:"SyncInfo"`
+		}
+
+		s.Require().Eventually(
```

Comment:
Should we be doing this check for all validators in a for loop?

Reply:
Correct, will do